app/src/controllers/user/sagas.js

```diff
@@ -150,19 +151,44 @@ function* fetchUserWorker() {
     ? assignedOrganizations[defaultOrganization]
     : Object.keys(assignedOrganizations)[0];
 
+  const isSavedOrganizationExist =
+    savedOrganizationSlug && savedOrganizationSlug in assignedOrganizations;
+
   const isSavedProjectExist =
-    savedOrganizationSlug &&
-    savedOrganizationSlug in assignedOrganizations &&
-    savedProjectSlug &&
-    savedProjectSlug in assignedProjects;
+    savedProjectSlug && savedProjectSlug in assignedProjects && isSavedOrganizationExist;
+
+  const isAdmin = userRole === ADMINISTRATOR;
+  const isManager = assignedOrganizations?.[savedOrganizationSlug]?.organizationRole === MANAGER;
 
-  const activeProject = isSavedProjectExist
-    ? savedActiveProject
-    : { organizationSlug: defaultOrganizationSlug, projectSlug: defaultProjectSlug };
+  let projectKey;
 
-  const projectKey = isSavedProjectExist
-    ? assignedProjects[savedProjectSlug].projectKey
-    : defaultProjectKey;
+  if (!isSavedProjectExist && (isAdmin || (isManager && isSavedOrganizationExist))) {
+    try {
+      const activeOrganization = yield call(
+        fetch,
+        URLS.organizationList({ slug: savedOrganizationSlug }),
+      );
+      const id = activeOrganization?.items?.[0]?.id;
+
+      // TODO: Fetch project by slug
+      const organizationProjects = yield call(fetch, URLS.organizationProjects(id));
+      projectKey = organizationProjects?.items?.find(({ slug }) => slug === savedProjectSlug)?.key;
+    } catch (err) {
+      yield put(fetchUserErrorAction());
```

Comment:
I'm not sure it is the right action here as we've fetched the user successfully.
I suppose in case of error the default project/org should be used.

Reply:
Done.